mesonbuild/mcompile.py

```diff
@@ -227,6 +227,12 @@ def get_parsed_args_vs(options: 'argparse.Namespace', builddir: Path) -> T.List[
 
     cmd += options.vs_args
 
+    # Add platform from sln file because msbuild assumes that for 32-bit build
+    # the platform is x86 when it should be Win32. The platform is extracted from first
+    # line that contains buildtype|platform e.g. debug|Win32
+    platform = re.search('(?<=(\|))[a-zA-Z0-9]*(?=(\n))', open(sln, 'r').read()).group()
+    cmd.append('-p:Platform=' + platform)
```

Comment:
The error you're seeing is because the visual studio command prompt sets the `Platform` env var to `x64` for 64-bit and `x86` for 32-bit, and MSBuild picks it up. The correct solution here would be to unset that before calling `msbuild`. Our solutions only contain a single `config|arch` anyway.

Reply:
Looking at the code, the best way is probably to return `cmd, env` from all `get_parsed_args_*()` functions and pass it via the `env=` kwarg to `Popen_safe`.

`get_parsed_args_ninja()` would return `cmd, None` and `gst_parsed_args_vs()` would return `cmd, env` where `env = os.environ.copy() \n del env['Platform']` or similar.

Reply:
Changed as suggested, seems to work fine. CI build would probably be nice because test scripts still do not work with 32-bit dev prompt since they call msbuild directly rather than "meson compile".